packages/fast-components-styles-msft/src/utilities/density.ts

```diff
@@ -52,15 +53,41 @@ export function heightNumber(lines: number = 1): DesignSystemResolver<number> {
  * @param designSystem The design system config.
  */
 export function getDensityCategory(designSystem: DesignSystem): DensityCategory {
+    const densityValue: number = getDesignSystemValue("density")(designSystem);
     const category: DensityCategory =
-        designSystem.density >= 2
+        densityValue >= 2
             ? DensityCategory.spacious
-            : designSystem.density <= -2
+            : densityValue <= -2
                 ? DensityCategory.compact
                 : DensityCategory.normal;
     return category;
 }
 
+/**
+ * Returns a value based on the higher-level category for the density setting.
+ * Used to adjust things like type size and sizing that is based on the category rather than individual density.
+ *
+ * @param compactValue The adjustment when the category is "compact"
+ * @param spaciousValue The adjustment when the category is "spacious"
+ * @param normalValue The adjustment when the category is "normal"
+ */
+export function densityCategorySwitch<T = number>(
+    compactValue: T | DesignSystemResolver<T>,
+    spaciousValue: T | DesignSystemResolver<T>,
```

Comment:
I think that, because these are all required now, the argument order should be:
1. compact
2. normal
3. spacious